resources/packs/ms365/msgraph.go

```diff
@@ -9,17 +9,32 @@ import (
 	"github.com/microsoftgraph/msgraph-beta-sdk-go/models"
 	"github.com/microsoftgraph/msgraph-beta-sdk-go/rolemanagement/directory/roleassignments"
 	"go.mondoo.com/cnquery/motor/providers"
-	ms365_provider "go.mondoo.com/cnquery/motor/providers/ms365"
-	"go.mondoo.com/cnquery/motor/providers/ms365/msgraphclient"
-	"go.mondoo.com/cnquery/motor/providers/ms365/msgraphconv"
+	"go.mondoo.com/cnquery/motor/providers/microsoft/azure"
+	ms365_provider "go.mondoo.com/cnquery/motor/providers/microsoft/ms365"
+	"go.mondoo.com/cnquery/motor/providers/microsoft/ms365/msgraphclient"
+	"go.mondoo.com/cnquery/motor/providers/microsoft/ms365/msgraphconv"
+	"go.mondoo.com/cnquery/motor/vault"
 	"go.mondoo.com/cnquery/resources"
 	"go.mondoo.com/cnquery/resources/packs/core"
 )
 
-func ms365Provider(t providers.Instance) (*ms365_provider.Provider, error) {
+func msGraphProvider(t providers.Instance) (*ms365_provider.Provider, error) {
 	at, ok := t.(*ms365_provider.Provider)
 	if !ok {
-		return nil, errors.New("ms365 resource is not supported on this transport")
```

Comment:
it would read easier if we would use the go type switches here https://go.dev/tour/methods/16